lib/ansible/modules/pip.py

```diff
@@ -611,6 +613,64 @@ def setup_virtualenv(module, env, chdir, out, err):
     return out, err, cmd
 
 
+def _resolve_package_names(
+        module: AnsibleModule,
+        package_list: list[Package],
+        pip: list[str]
+) -> list[Package]:
+    """Return a list of Packages with names extracted from metadata using pip"""
+    pkgs_to_resolve = [pkg for pkg in package_list if not pkg.has_requirement]
+
+    if not pkgs_to_resolve:
+        return package_list
+
+    # Unsetting 'FORCE_COLOR' with 'NO_COLOR' and 'TTY_COMPATIBLE' env vars, helps to
+    # make pip output clean JSON without injected ANSI sequences that might corrupt it.
+    os.environ.pop('FORCE_COLOR', None)
+
+    rc, json_out, err = module.run_command(
+        [*pip, 'install', '--dry-run', '--ignore-installed', '--quiet', '--report=-', *(str(pkg) for pkg in pkgs_to_resolve)],
+        environ_update={
+            'NO_COLOR': '1',
+            'TTY_COMPATIBLE': '0',
+        },
+    )
+
+    if rc == 0:
+        # If it succeeds, handle the report
+        report = json.loads(json_out)
+        package_objects = (Package(install_report['metadata']['name'], version_string=install_report['metadata']['version'])
+                           for install_report in report['install'])
```

Comment:
This type of indentation isn't very common in Python and is known to cause larger diffs on changes. Plus, you have a chance to have shorter lines like this:
```suggestion
        package_objects = (
            Package(install_report['metadata']['name'], version_string=install_report['metadata']['version'])
            for install_report in report['install']
        )
```

Long lines are rather problematic with vertical tabs/split views/diffs in editors and various UIs. Especially when one uses larger font sizes to view this text. So it's an accessibility hazard, when the lines are overly long. In fact, text in printed media appears in columns because it's easier for humans to read. Typography studies show that columns should have around 50-75 CPL[^1]. While code is a little bit more nuanced, it's still good to have short lines with rare practical exceptions[^2].

[^1]: https://baymard.com/blog/line-length-readability
[^2]: https://blog.glyph.im/2025/08/the-best-line-length.html